client/app/lib/hooks/session.ts

```diff
@@ -71,3 +82,7 @@ export const setI18nConfig = (config: Partial<I18nConfig>): void => {
 
   imperativeDispatch(actions.setI18nConfig(config));
 };
+
+export const syncSignals = (signals: Record<string, number>): void => {
```

Comment:
Would it be better to restrict the type of the signals keys to `announcements`, `assessment_submissions`, etc instead of string?

Reply:
With the way the framework is currently written, the pipeline for syncing signals is only responsible for piping the signals through. It could care less about the actual payload it's piping, i.e., they can pipe any signals as long as the "shape" of the signal conforms to the framework's specification (and that is currently just a record). To keep parts of the pipeline closed to modifications on extensions of other different signals, we should keep them oblivious to the concrete payload's contents.

Then again, `announcements`, `assessment_submissions`, etc. are not signal keys. They are just the sidebar's keys hard-defined in `sidebar_items` in `*_component.rb`. For now, we don't have any signal keys yet, i.e., signals do not emit their own type. But if such needs arise for other use cases, we should have them.